lib/src/cpp/utils/XMLManager.cpp

```diff
@@ -69,42 +77,39 @@ XMLManager::XMLManager (
     // Parser would validate the serialized XML against the defined XSD, with some configured parameters
     parser = new xercesc::XercesDOMParser();
 
-    // Error handler would receive the exception from the parser, and report it as a FileNotFound expected exception
+    // Error handler would receive the exception from the parser, and report it as a ElementNotFound exception
     error_handler = new utils::ErrorHandlerXMLManager(
-        utils::ErrorHandlerXMLManager::Kind::FileNotFound);
+        utils::ErrorHandlerXMLManager::Kind::ElementNotFound);
     parser->setErrorHandler(error_handler);
 
     // Read XML file and obtain the doc object
     try
     {
         // Parsing initially would load the elements in the parser's doc object
         parser->parse(xml_file.c_str());
+
         // Doc would be needed in some other process, so it would be useful to take it apart of the parser
         doc = parser->getDocument();
+
+        // Obtain first document node
+        reference_node = static_cast<xercesc::DOMNode*>(doc->getDocumentElement());
     }
     // Given file does not exist
-    catch (const FileNotFound& ex)
+    catch (const ElementNotFound& ex)
     {
         // Create new document if required
-        if (create_file && !file_exists)
+        if (!file_exists)
```

Comment:
If we are checking this flag, then does it mean that the exception can be thrown by other cause? If so, then the comment in the catch is not correct.

Reply:
The file may exist but be empty, or at least not be parseable.
In that case, doc would not be created and with the current implementation, it would cause `SEG_FAULT`.
We should, or add an ´else´ sentence that throws an exceptions, or remove the `if` sentence.